samples/matter/lock/src/app_task.cpp

```diff
@@ -274,6 +288,71 @@ void AppTask::LockActionEventHandler(const AppEvent &event)
 	}
 }
 
+#ifdef CONFIG_THREAD_WIFI_SWITCHING
+void AppTask::SwitchModesEventHandler(const AppEvent &event)
+{
+	/* Swap the application from using Thread to Wi-Fi or opposite. */
+#if defined(CONFIG_NET_L2_OPENTHREAD)
+	LOG_INF("Switching application from Thread to Wi-FI triggered.");
+	SystemLayer().ScheduleLambda([] {
```

Comment:
I am wondering if this is justified to run the swap asynchronously. It seems to be safer to use it as a blocking call as we don't expect any asynchronous notifications from the swapper module anyway.

Reply:
Maybe you are right. I did it mainly, because according to task description we wanted to run the swap on separate thread, but I think now it will block other Matter events anyway. The only reason is that swap will be invoked from the system events queue and not directly in this event handler.